components/GridView.vue

```diff
@@ -116,26 +108,39 @@ async function validateAndSaveLayout(newLayout: string): Promise<void> {
   console.log('Layout saved 🎉')
 }
 
+function resizeHandler(): void {
+  if (resizeTimeout) clearTimeout(resizeTimeout)
+  resizeTimeout = setTimeout(() => {
+    const prevCols = gridColumns.value
+    const newCols = getGridColumns(window.innerWidth)
+
+    if (prevCols !== newCols) {
+      gridColumns.value = newCols
+      layout.value = toGridLayoutItems(layout.value, newCols)
+    }
+  }, DEBOUNCE_TIMEOUT)
+}
+
 function resetLayout(): void {
   showSettingsModal.value = false
   const newUserLayout = getNewUserLayout(address)
   layout.value = toGridLayoutItems(newUserLayout, COL_NUM_LARGE)
 }
 
-function breakpointChanged(
-  _newBreakpoint: Breakpoint,
-  _newLayout: Layout
-): void {
-  triggerLayoutRefresh()
-}
-
 function clearSelection(): void {
   window.getSelection()?.removeAllRanges()
 }
 
+onMounted(() => {
+  window.addEventListener('resize', resizeHandler)
```

Comment:
Might be better to use https://vueuse.org/core/useResizeObserver/#useresizeobserver but we can add that later.

Reply:
👍 will update in a follow up pr. Will merge this one so the feature test pr gets updated